### PR TITLE
[Finder] Get filename without extension

### PR DESCRIPTION
**Pull request by antonch1989**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes 
| Fixed tickets | #30655 
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/Finder/SplFileInfo.php [hunk 1]
```diff
@@ -57,6 +57,18 @@ public function getRelativePathname()
         return $this->relativePathname;
     }
 
+    /**
+     * Returns the filename without an extension.
+     *
+     * @return string
```
Comment (OskarStark) [inline src/Symfony/Component/Finder/SplFileInfo.php]: Please remove this line and add a return type to the method

**Review by OskarStark — COMMENTED** [review]
Thank you for this contribution, please fix the table in the PR header, because this is no bugfix, but a feature.

File: src/Symfony/Component/Finder/SplFileInfo.php [hunk 2]
```diff
@@ -57,6 +57,16 @@ public function getRelativePathname()
         return $this->relativePathname;
     }
 
+    /**
+     * Returns the filename without an extension.
+     */
```
Comment (fabpot) [inline src/Symfony/Component/Finder/SplFileInfo.php]: I would remove the whole block as the method name is quite descriptive already.

**Review by fabpot — APPROVED** [review]
LGTM, let's a minor comment.

**Comment by fabpot** [comment]
Thank you @antonch1989.

**Comment by OskarStark** [comment]
Thank you for this contribution, can you please add some documentation for this new feature?

**Comment by javiereguiluz** [comment]
I find this method confusing. It's called `getFilenameWithoutExtension()` but sometimes it returns the filename + extension:

```php
getFilenameWithoutExtension('foo.html');      // foo
getFilenameWithoutExtension('foo.html.twig'); // foo.html
getFilenameWithoutExtension('foo.tar.gz');    // foo.tar
```

I expected `foo` as the returned value in all cases.

**Pull request merged**